projects/storefrontlib/src/cms-components/cart/add-to-cart/add-to-cart.component.ts

```diff
@@ -43,16 +55,41 @@ export class AddToCartComponent implements OnInit, OnDestroy {
     quantity: new FormControl(1, { updateOn: 'blur' }),
   });
 
+  constructor(
+    modalService: ModalService,
+    currentProductService: CurrentProductService,
+    cd: ChangeDetectorRef,
+    activeCartService: ActiveCartService,
+    componentData: CmsComponentData<model>
+  );
+
+  /**
+   * @deprecated since 3.4
+   */
+  constructor(
+    modalService: ModalService,
+    currentProductService: CurrentProductService,
+    cd: ChangeDetectorRef,
+    activeCartService: ActiveCartService
+  );
+
   constructor(
     protected modalService: ModalService,
     protected currentProductService: CurrentProductService,
-    protected cd: ChangeDetectorRef,
-    protected activeCartService: ActiveCartService
+    private cd: ChangeDetectorRef,
+    protected activeCartService: ActiveCartService,
+    @Optional() protected componentData?: CmsComponentData<model>
```

Comment:
If I get a better argument to not make it configurable using the `ViewConfig` to make it toggle-able, then 

 ```suggestion
    @Optional() protected componentData?: CmsComponentData<CmsAddToCartComponent>
```

Reply:
this line of code is no longer in use.

Reply:
```suggestion
    @Optional() protected component?: CmsComponentData<CmsAddToCartComponent>
```

note the change of name and the typing. Instead of putting an alias as model, put it straight as CmsAddToCartComponent